sectionproperties/analysis/cross_section.py

```diff
@@ -3710,6 +3817,155 @@ def plot_stress_vm(self, pause=True, cmap="coolwarm", normalize=True, size=500,
+    
+    
+    def plot_mohrs_circles(self,x,y,pause=True, cmap="coolwarm", size=500, dpi=96):
+        """Plots Mohr's Circles of the 3D stress state at position x,y
+
+        :params x,y: Coordinates of the point to draw Mohr's Circle
+        :type x,y: float
+        :param bool pause: If set to true, the figure pauses the script until the window is closed.
+            If set to false, the script continues immediately after the window is rendered.
+
+        :return: Matplotlib figure and axes objects (fig, ax)
+        :rtype: (:class:`matplotlib.figure.Figure`, :class:`matplotlib.axes`)
+        
+        The following example plots the Mohr's Circles for the 3D stress state within a 150x90x12 UA section
+        resulting from the following actions:
+
+        * :math:`N = 50` kN
+        * :math:`M_{xx} = -5` kN.m
+        * :math:`M_{22} = 2.5` kN.m
+        * :math:`M_{zz} = 1.5` kN.m
+        * :math:`V_{x} = 10` kN
+        * :math:`V_{y} = 5` kN
+        
+        at the point (10,88.9)
+
+        ::
+
+            import sectionproperties.pre.sections as sections
+            from sectionproperties.analysis.cross_section import Section
+
+            geometry = sections.angle_section(d=150, b=90, t=12, r_r=10, r_t=5, n_r=8)
+            mesh = geometry.create_mesh(mesh_sizes=[2.5])
+            section = Section(geometry, mesh)
+
+            section.calculate_geometric_properties()
+            section.calculate_warping_properties()
+            stress_post = section.calculate_stress(
+                N=50e3, Mxx=-5e6, M22=2.5e6, Mzz=0.5e6, Vx=10e3, Vy=5e3
+            )
+            
+            stress_post.plot_mohrs_circles(10,88.9)
+
+        ..  figure:: ../images/stress/mohrs_circles.png
+            :align: center
+            :scale: 75 %
+
+            Mohr's Circles of the 3D stress state at (10,88.9).
+        
+        """
+        
+        pt = (x,y)
+        nodes = self.cross_section.mesh_nodes
+        ele = self.cross_section.mesh_elements        
+        triang = tri.Triangulation(nodes[:, 0], nodes[:, 1], ele[:, 0:3])
+        
+        # Find in which material group the point lies
+        pt_group = None
+        for group in self.material_groups:            
+            mask_array = np.ones(len(self.cross_section.elements), dtype=bool)
+            mask_array[group.el_ids] = False
+            triang.set_mask(mask_array)
+            trifinder = triang.get_trifinder()
+            if trifinder(*pt) != -1:
+                pt_group = group
+            triang.set_mask(None)
+        
+        if pt_group is None:
+            raise Exception(f"Point {(*pt,)} is not within mesh")
+        
+        # Assesmble the stress results from the relevant material group
+        sigma_zz_v = pt_group.stress_result.sig_zz
+        tau_xz_v = pt_group.stress_result.sig_zx
+        tau_yz_v = pt_group.stress_result.sig_zy
+            
+        # Get the interpolators
```

Comment:
Technically the triangular elements used have quadratic shape functions so higher order interpolation can be used. I think for the purposes of this application, linear interpolation will be acceptable. May give inaccurate results for coarse mesh in regions of high stress gradient?

Reply:
See #68

Reply:
@robbievanleeuwen sure thing! Will definitely add to the theory section in the docs. Your pdf referenced in #68 is basically the complete theory, especially when backed up with Pilkey. 

And it's a great observation about the interpolations used. It might be better to abstract that bit out to another new API function that can then be implemented appropriate to the element type used, and available to users.